exporters/metrics/src/main/java/com/google/cloud/opentelemetry/metric/MetricTranslator.java

```diff
@@ -104,26 +104,59 @@ static MetricDescriptor mapMetricDescriptor(
         .forEach((key, value) -> builder.addLabels(mapAttribute(key, value)));
 
     MetricDataType metricType = metric.getType();
-    if (GAUGE_TYPES.contains(metricType)) {
+    switch (metricType) {
+      case LONG_GAUGE:
+        builder.setMetricKind(MetricDescriptor.MetricKind.GAUGE);
+        builder.setValueType(MetricDescriptor.ValueType.INT64);
+        return builder.build();
+      case DOUBLE_GAUGE:
+        builder.setMetricKind(MetricDescriptor.MetricKind.GAUGE);
+        builder.setValueType(MetricDescriptor.ValueType.DOUBLE);
+        return builder.build();
+      case LONG_SUM:
+        builder.setValueType(MetricDescriptor.ValueType.INT64);
+        return fillSumType(metric.getLongSumData(), builder);
+      case DOUBLE_SUM:
+        builder.setValueType(MetricDescriptor.ValueType.DOUBLE);
+        return fillSumType(metric.getDoubleSumData(), builder);
+      case HISTOGRAM:
+        return fillHistogramType(metric.getDoubleHistogramData(), builder);
+      default:
+        logger.error(
+            "Metric type {} not supported. Only gauge and cumulative types are supported.",
+            metricType);
+    }
+    return null;
+  }
+
+  private static MetricDescriptor fillHistogramType(
+      DoubleHistogramData histogram, MetricDescriptor.Builder builder) {
+    builder.setValueType(MetricDescriptor.ValueType.DISTRIBUTION);
+    switch (histogram.getAggregationTemporality()) {
+      case CUMULATIVE:
+        builder.setMetricKind(MetricDescriptor.MetricKind.CUMULATIVE);
+        return builder.build();
+      default:
+        logger.error(
+            "Histogram type {} not supported. Only cumulative types are supported.", histogram);
```

Comment:
Will this log the DELTA part which is the unsupported bit?

Reply:
yes